.gitlab-ci.yml

```diff
@@ -86,3 +102,23 @@ publish:s3:
       done
   only:
     - master
+
+publish:s3:latest:
+  only:
+    variables:
+      - $MENDER_VERSION =~ /[0-9]+\.[0-9]+\.[0-9]+/
+  stage: publish
+  image: debian:buster
+  dependencies:
+    - build
+  before_script:
+    - apt update && apt install -yyq awscli
+  script:
+    - echo "Publishing latest packages to S3"
+    - deb_version=$(cat output/mender-client-deb-version)
+    - for arch in amd64 arm64 armhf; do
+        aws s3 cp output/mender-client_${deb_version}_${arch}.deb
+          s3://${S3_BUCKET_NAME}/latest/${S3_BUCKET_SUBPATH}/${arch}/mender-client_${deb_version}_${arch}.deb;
+        aws s3api put-object-acl --acl public-read --bucket ${S3_BUCKET_NAME}
+          --key latest/${S3_BUCKET_SUBPATH}/${arch}/mender-client_${deb_version}_${arch}.deb;
+      done
```

Comment:
To reuse the code, we could move `deb_version=...` to `before_script` in `publish:s3`, and then make this pipeline job to extend `publish:s3`, by only overriding the `before_script`.